src/components/new-suggestion/areas/instruments/instruments.area.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from "react"
+import InstrumentsEdit from "./edit/instruments-edit"
+import InstrumentSearch from "@/components/search/instrument/instrument-search"
+import { Instrument } from "@/types/database-types"
+import { setActiveArea, updateNewSuggestion } from "@/redux/slices/new-suggestion.slice"
+import { AppDispatch, AppState } from "@/redux/store"
+import { useDispatch, useSelector } from "react-redux"
+import { NewSuggestionInstrument } from "@/interfaces/new-suggestion"
+import { Area } from "@/constants/area"
+
+interface InstrumentsAreaProps {
+  instruments: Instrument[]
+}
+
+const InstrumentsArea = ({ instruments }: InstrumentsAreaProps) => {
+  const dispatch: AppDispatch = useDispatch()
+  const newSuggestion = useSelector((state: AppState) => state.newSuggestion.suggestion)
+  const [canProceed, setCanProceed] = useState<boolean>(newSuggestion.instruments.length > 0)
+
+  useEffect(() => {
+    setCanProceed(newSuggestion.instruments.length > 0)
```

Comment:
No need to put this in useEffect. You could even put this in the disabled directly. This is because 'newSuggestion' is a hook in itself.